idk/ingest.go

```diff
@@ -1121,189 +1131,386 @@ func (m *Main) runDeleter(c int, limitCounter *msgCounter) error {
-				if err != nil {
-					return errors.Wrap(err, "clearing set")
+				m.log.Debugf("Delete consumer running the follow delete queries: %s", bq.Serialize())
+
+				resp, err := client.Query(bq, nil)
+				if err != nil || resp.Success != true {
+					return errors.Wrap(err, "error deleting values")
 				}
-				CounterDeleterRowsAdded.With(prom.Labels{"type": "set"}).Inc()
-			case pilosaclient.FieldTypeMutex:
-				if val == "" {
-					continue
+			case "records":
+				// deleting a record
+				var rawQueries []string
+				if fieldValues["keys"] != nil {
+					// if keys set, delete list of record keys
+					keysAsStrings, err := toStringArray(fieldValues["keys"])
+					if err != nil {
+						return errors.Errorf("unable to convert 'keys' value to an array of strings")
+					}
+					columnKeys := "'" + strings.Join(keysAsStrings, "','") + "'"
+					rawQueries = append(rawQueries, fmt.Sprintf("Delete(ConstRow(columns=[%s]))", columnKeys))
+
 				}
-				_, err := client.Query(index.BatchQuery(
-					field.Clear(val, recordID),
-				))
-				if err != nil {
-					return errors.Wrap(err, "clearing mutex")
-				}
-				CounterDeleterRowsAdded.With(prom.Labels{"type": "mutex"}).Inc()
-			case pilosaclient.FieldTypeBool:
-				_, err := client.Query(index.BatchQuery(
-					field.Clear(0, recordID),
-					field.Clear(1, recordID),
-				))
-				if err != nil {
-					return errors.Wrap(err, "clearing bool")
+				if fieldValues["ids"] != nil {
+					// if ids set, delete list of record IDs
+					idsAsInts, err := toUint64Array(fieldValues["ids"])
+					if err != nil {
+						return errors.Errorf("unable to convert 'ids' value to an array of int64s")
+					}
+					keysAsStrings := make([]string, len(idsAsInts))
+					for i, v := range idsAsInts {
+						keysAsStrings[i] = fmt.Sprint(v)
+					}
+					columnKeys := strings.Join(keysAsStrings, ",")
+					rawQueries = append(rawQueries, fmt.Sprintf("Delete(ConstRow(columns=[%s]))", columnKeys))
 				}
-				CounterDeleterRowsAdded.With(prom.Labels{"type": "bool"}).Inc()
-			case pilosaclient.FieldTypeInt:
-				_, err := client.Query(field.Clear(0, recordID))
-				if err != nil {
-					return errors.Wrap(err, "clearing int")
+				if fieldValues["filter"] != nil {
+					// if filter set, use it as filter in delete query
+					filter, ok := fieldValues["filter"].(string)
+					if !ok {
+						return errors.Errorf("unable to convert fitler into string")
+					}
+					rawQueries = append(rawQueries, fmt.Sprintf("Delete(%s)", filter))
+				}
+				if len(rawQueries) == 0 {
+					m.log.Infof("delete record doesn't contain any delete queries: confirm 'keys', 'ids', or 'filter' key has a value")
+				}
+				for _, query := range rawQueries {
+					baseQuery := index.RawQuery(query)
+					bq.Add(baseQuery)
 				}
-				CounterDeleterRowsAdded.With(prom.Labels{"type": "int"}).Inc()
-			case pilosaclient.FieldTypeDecimal:
-				_, err := client.Query(field.Clear(0, recordID))
+
+				_, err := client.Query(bq, nil)
```

Comment:
from what i can gather if len(rawQueries) ==0, there is no need to execute the client query, maybe put this in an else clause, Note I have no idea why github decided to plop those deleted lines in this remark, i only intended to point at line 1498